CauldronMods/Controller/Villains/TheInfernalChoir/Cards/VagrantHeartPhase1CardController.cs

```diff
@@ -0,0 +1,174 @@
+            base.AddTrigger<MakeDecisionsAction>((MakeDecisionsAction md) => md.CardSource != null && !md.CardSource.Card.IsVillain, this.RemoveDecisionsFromMakeDecisionsResponse, TriggerType.RemoveDecision, TriggerTiming.Before);
+
+            base.AddTriggers();
+        }
+
+        private bool HeartMoveCriteria(MoveCardAction mca)
+        {
+            return mca.WasCardMoved && mca.Origin.OwnerTurnTaker == VagrantTurnTaker && !mca.Destination.IsRevealed && !VagrantDeck.HasCards;
+        }
+
+        private bool HeartPlayCriteria(PlayCardAction pca)
+        {
+            return pca.WasCardPlayed && pca.CardToPlay.Owner == VagrantTurnTaker && !VagrantTurnTaker.Revealed.HasCards && !VagrantDeck.HasCards;
+        }
+
+        private bool HeartDrawCriteria(DrawCardAction dca)
+        {
+            return dca.DidDrawCard && dca.HeroTurnTaker == VagrantTurnTaker && !VagrantDeck.HasCards;
+        }
+
+        private bool HeartBulkMoveCriteria(BulkMoveCardsAction bmca)
+        {
+            return !VagrantTurnTaker.Revealed.HasCards && !VagrantDeck.HasCards;
+        }
+
+        /* ..shuffle all cards under Vagrant Heart back into the hero's deck, and flip {TheInfernalChoir}'s villain character cards and Vagrant Heart." */
+        private IEnumerator GoToPhase2(GameAction action)
+        {
+            var coroutine = GameController.FlipCard(CharacterCardController, actionSource: action, cardSource: CharacterCardController.GetCardSource(), allowBackToFront: false);
+            if (UseUnityCoroutines)
+            {
+                yield return GameController.StartCoroutine(coroutine);
+            }
+            else
+            {
+                GameController.ExhaustCoroutine(coroutine);
+            }
+        }
+
+        private IEnumerator VagrantHeartGameOver(TargetLeavesPlayAction action)
+        {
+            var coroutine = GameController.GameOver(EndingResult.AlternateDefeat, $"The Vagrant Heart consumes {VagrantTurnTaker.Name}! [b]The heroes are defeated.[/b]", actionSource: action, cardSource: GetCardSource());
+            if (UseUnityCoroutines)
+            {
+                yield return GameController.StartCoroutine(coroutine);
+            }
+            else
+            {
+                GameController.ExhaustCoroutine(coroutine);
+            }
+        }
+
+        protected IEnumerator VagrantHeartDamageReponse(DealDamageAction action)
+        {
+            IEnumerator coroutine;
+            int amount = action.Amount; //NB: ImmuneToDamage call reduces amount to zero, store value ahead of time.
+
+            coroutine = GameController.ImmuneToDamage(action, GetCardSource());
+            if (UseUnityCoroutines)
+            {
+                yield return GameController.StartCoroutine(coroutine);
+            }
+            else
+            {
+                GameController.ExhaustCoroutine(coroutine);
+            }
+
+            var vagrantHero = Card.Location.OwnerTurnTaker;
```

Comment:
Why aren't you just calling your VagrantTurnTaker property here? It would seem to do the same thing without the redundancy?

Reply:
Made the switch and it did not seem to break anything